formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Streams.kt

```diff
@@ -43,12 +43,18 @@ internal class ByteArrayOutput {
     private var position: Int = 0
 
     private fun ensureCapacity(elementsToAppend: Int) {
-        if (position + elementsToAppend <= array.size) {
+        val requiredCapacityLong = position.toLong() + elementsToAppend.toLong()
+        if (requiredCapacityLong > Int.MAX_VALUE) {
+            throw IllegalArgumentException("Required capacity exceeds maximum array size (Int.MAX_VALUE).")
+        }
+
+        val requiredCapacity = requiredCapacityLong.toInt()
+        if (requiredCapacity <= array.size) {
             return
         }
-        val newArray = ByteArray((position + elementsToAppend).takeHighestOneBit() shl 1)
-        array.copyInto(newArray)
-        array = newArray
+
+        val newCapacity = nextPowerOfTwoCapacity(requiredCapacity)
```

Comment:
```suggestion
        val newCapacity = (requiredCapacityLong.takeHighestOneBit() shl 1)
            .coerceAtMost(Int.MAX_VALUE.toLong()).toInt()
```

It seems like the only reason to have a dedicated function is to test it, but since the function is trivial, I would rather embed it here, and simplify this expression (note that we don't have to cast `requiredCapacityLong` to `int` anymore).

Reply:
I would not recommend this change as proposed, because it again **removes test coverage** for this logic. I understand the motivation for inlining, but this leaves a non-trivial piece of functionality without dedicated tests. **This lack of testing is precisely why the current serialization is faulty**: there are no tests, and this change would discard them again. A better long-term solution would avoid imposing such limits altogether, although that is outside the scope of my change.